### PR TITLE
Core/Spells: implement SMSG_MOUNT_RESULT and use it for transformed mounting cases

### PR DESCRIPTION
**Pull request by Ovahlord**
**Changes proposed:**

-  implement mount result opcode handling which got introduced with 4.x and use it in the currently only known sniffable case when trying to mount up while being transformed

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Tests performed:** (Does it build, tested in-game, etc.)
- tested on my 434 fork
WoWScrnShot_040720_184548

### CODE REVIEW

File: src/server/game/Miscellaneous/SharedDefines.h [hunk 1]
```diff
@@ -6508,4 +6508,19 @@ enum class GameError : uint32
     ERR_NOT_IN_PET_BATTLE                                           = 1047,
 };
 
+enum class MountResult : uint32
+{
+    InvalidMountee = 0,
+    TooFarAway     = 1,
+    AlreadyMounted = 2,
+    NotMountable   = 3,
+    NotYourPet     = 4,
```
Comment (Faq) [inline src/server/game/Miscellaneous/SharedDefines.h:6517]: In what cases this used?

File: src/server/game/Miscellaneous/SharedDefines.h [hunk 2]
```diff
@@ -6508,4 +6508,19 @@ enum class GameError : uint32
...
+    Other          = 5,
+    Looting        = 6,
```
Comment (Faq) [inline src/server/game/Miscellaneous/SharedDefines.h:6519]: How can Mount result be " Looting" ?

Comment (Ovahlord) [reply src/server/game/Miscellaneous/SharedDefines.h:6519]: These values are from wpp. These values are not only for smsg_mount_result but also for smsg_dismount_result so some of these values are used for the other opcode as well which I have not discovered in sniffs yet.

**Comment by ForesterDev** [comment]
Why in 4.x? It's present in 3.3.5 - SMSG_MOUNTRESULT

**Comment by Faq** [comment]
@Ovahlord Can you make PR also for 335?

File: src/server/game/Miscellaneous/SharedDefines.h (hunk 1 above)
Comment (illfated) [reply src/server/game/Miscellaneous/SharedDefines.h:6517]: Maybe parked mounts without vehicle access?

File: src/server/game/Miscellaneous/SharedDefines.h (hunk 2 above)
Comment (Shauren) [reply src/server/game/Miscellaneous/SharedDefines.h:6519]: Values are correct, comment is wrong. They are used exclusively for SMSG_MOUNT_RESULT.

**Pull request merged**